src/meta/app/src/principal/user_stage.rs

```diff
@@ -555,6 +555,7 @@ pub struct StageInfo {
     pub stage_name: String,
     pub stage_type: StageType,
     pub stage_params: StageParams,
+    pub from_uri: bool,
```

Comment:
It seems we no need add the from_uri, we can check it by add a new method for the StageInfo:
impl StageInfo {
  pub fn uri_kind() -> URIKind{}
}